src/context.cpp

```diff
@@ -640,9 +643,11 @@ boost::optional<const actor_t&>
 context_t::locate(const std::string& name) const {
     auto ptr = m_services.synchronize();
     auto it  = std::find_if(ptr->begin(), ptr->end(), match{name});
-
+    if(it == ptr->end()) {
+        return boost::none;
+    }
     return boost::optional<const actor_t&>(
-        it != ptr->end() && it->second->is_active(),
```

Comment:
Yeah, that's what I meant — if something calls `locate()`, stores the reference somewhere and then some other codepath removes the service, the reference will be invalid. It was never an issue though, but still is an UB. It might be fixed with `weak_ptr<T>`, but I'm not sure if it won't break anything else, can you investigate? As for this PR, I'd suggest dropping that part because it's irrelevant here and moving it to a new PR.